### PR TITLE
[Serverless] Add java integration tests for the Serverless extension

### PR DESCRIPTION
**Pull request by jcstorms1**
### What does this PR do?

Adds Java Integration tests for the Serverless extension

### Motivation
Improves testing coverage

### Describe how to test/QA your changes
Run the integration tests in the Serverless folder

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/logs/logs_test.go [hunk 1]
```diff
@@ -471,7 +471,7 @@ func TestProcessLogMessageNoStringRecordFunctionLog(t *testing.T) {
 
 	logMessages := []logMessage{
 		{
-			logType: logTypeFunction,
+			stringRecord: "hi, log 2",
```
Comment (nhinsch) [inline pkg/serverless/logs/logs_test.go:474]: The name of this test contains `NoStringRecord` but it looks like you added a `stringRecord` to the test data here. What was the motivation for this change? Also, is this related to adding Java integration tests?

Comment (nhinsch) [reply pkg/serverless/logs/logs_test.go:474]: Per convo, this might be related to the fact that we haven't merged the v16 changes back into the main branch yet -- I'll wait for that to be merged before reviewing further.

File: pkg/trace/agent/agent.go [hunk 2]
```diff
@@ -491,3 +491,12 @@ func newEventProcessor(conf *config.AgentConfig) *event.Processor {
 func (a *Agent) SetGlobalTagsUnsafe(tags map[string]string) {
 	a.conf.GlobalTags = tags
 }
+
```
Comment (maxday) [inline pkg/trace/agent/agent.go:494]: you can discard this change, this is no longer require as this function is now inlined (ref : https://github.com/DataDog/datadog-agent/pull/10035/files#diff-c2dc8d00a060f89f4cd810fa2496eb721f7da1a6f538c4b2ec5c5f7a894a76e7R253)

File: test/integration/serverless/recorder-extension/main.go [hunk 3]
```diff
@@ -247,14 +247,14 @@ func startHTTPServer(port string) {
...
+		pl := new(pb.AgentPayload)
 		if err := pl.Unmarshal(decompressedBody); err != nil {
 			fmt.Printf("Error while unmarshalling traces %s \n", err)
 			return
 		}
-		for _, trace := range pl.Traces {
-			jsonTrace, err := json.Marshal(trace)
+
+		for _, trace := range pl.TracerPayloads {
+			jsonTrace, err := json.Marshal(trace.Chunks)
```
Comment (nhinsch) [inline test/integration/serverless/recorder-extension/main.go:257]: What does this do?

File: test/integration/serverless/recorder-extension/go.mod [hunk 4]
```diff
@@ -3,6 +3,7 @@ module datadog-lambda-extension/recorder-extension
 go 1.16
 
 replace (
+	github.com/DataDog/datadog-agent/pkg/obfuscate => github.com/DataDog/datadog-agent/pkg/obfuscate v0.0.0-20211201172000-1fd9a353e8e4
```
Comment (nhinsch) [inline test/integration/serverless/recorder-extension/go.mod:6]: What is this for?

File: test/integration/serverless/src/logPython.py [hunk 5]
```diff
@@ -1,4 +1,3 @@
-# flake8: noqa
```
Comment (nhinsch) [inline test/integration/serverless/src/logPython.py:1]: Just a heads up, removing these comments means that the Python test functions are going to need to comply with the agent's Python linter. It's not necessarily a bad thing, but you might have to make some more changes to these files to get it passing (or you could put the comments back).

File: test/integration/serverless/src/java-tests/log/target/maven-archiver/pom.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#Generated by Maven
+#Mon Nov 29 10:50:44 EST 2021
+groupId=com.integration-test
+artifactId=log
+version=dev
```
Comment (nhinsch) [inline test/integration/serverless/src/java-tests/log/target/maven-archiver/pom.properties:5]: I am not familiar with Maven or the Java ecosystem in general, but do we need to commit all of these files? Generally files that are generated by a build do not need to be committed in the repository because they will be regenerated when someone runs the tests.

Comment (jcstorms1) [reply test/integration/serverless/src/java-tests/log/target/maven-archiver/pom.properties:5]: no this should have been in git ignore
